### ISSUE TITLE
Bug: [no-invalid-void-type] Does not allow using void as a default type for a generic argument

### ISSUE THREAD

**Issue by bogdanb** [typescript-eslint/typescript-eslint#5644]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=4.8.3&sourceType=module&code=KYDwDg9gTgLgBASwHY2FAZgQwMbDgUREwFswAbYAHgBU4BeOANwgQBMA+OAbwFgAoOHGLAYACwisAFAEoAXHGoBufgF9+QA&eslintrc=N4KABGBEAOCGBOBnApvSAuKABALgT2mUQGN4BLaHAWiIBsyA7HAejiVUgBpwp4BXWkQxhQECJFwEipCtTqMWDAPZVGAN1j0AJlTVKyO-IWEBtSKnhK03MWOBRNtJQHcAkgwDiyBqjLEAKlIAgvAA5nwAtt44iMI4-MgAvjwQALo8yclAA&tsconfig=N4KAvkA

### Repro Code

```typescript
export interface Example<T = void> {
  method(): T;
}
```

### ESLint Config

```javascript
{
  "parser": "@typescript-eslint/parser",
  "rules": {
    "@typescript-eslint/no-invalid-void-type": ["error",
      { "allowInGenericTypeArguments": true}
    ]
  }
}
```

### tsconfig

```
{}
```

### Expected Result

I expected the example code to pass linting without warnings or errors.

### Actual Result

On the first line we get the error _void is only valid as a return type or generic type variable_.

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `TypeScript`                           | `4.8.3` |
| `ESLint`                               | `8.15.0` |

**Comment by JoshuaKGoldberg** [comment]
I think it makes sense that if `allowInGenericTypeArguments` is specified, we should allow the default for a generic type variable to be `void` too. 👍

**Comment by kazizi55** [comment]
I would like to work on this issue!
